nntrainer/layers/layer_context.h

```diff
@@ -57,14 +57,16 @@ class InitLayerContext {
    * @brief Construct a new Init Layer Context object
    *
    */
-  InitLayerContext() = default;
+  InitLayerContext() : InitLayerContext({}, 1) {}
 
   /**
    * @brief Construct a new Init Layer Context object
    *
    * @param dim Input dimensions for the layer
    */
-  InitLayerContext(const std::vector<TensorDim> &dim) : input_dim(dim) {}
+  InitLayerContext(const std::vector<TensorDim> &dim, unsigned int num_out) :
+    input_dim(dim),
+    num_outputs(num_out) {}
```

Comment:
I guess these are layer node properties. Does the initcontext need to have those? is it because multi out and multi input?

Reply:
Yes, it is layer node property.
The number of outputs are fixed by the connections in the graph. Some nodes need that information such as split node.
Further for other nodes, setting the output dimensions should not exceed num_outputs already set inside InitLayerContext.